astropy/io/votable/tree.py

```diff
@@ -1170,7 +1172,7 @@ def __init__(self, votable, ID=None, name=None, datatype=None,
         # actually contains character data.  We have to hack the field
         # to store character data, or we can't read it in.  A warning
         # will be raised when this happens.
-        if (not config.get('pedantic') and name == 'cprojection' and
+        if (config.get('verify') == 'warn' and name == 'cprojection' and
```

Comment:
With this change, this `if` block is also skipped when `verify` is `'ignore'`, in which `datatype` and `arraysize` are also not set. Will this introduce a bug?

Reply:
Fixed